docs/modules/user-guide/partials/con_using-a-top-level-attribute.adoc

```diff
@@ -0,0 +1,38 @@
+[id="con_using-top-level-attribute_{context}"]
+= Using a top-level attribute
+
+[role="_abstract"]
+
+Use top-level attributes in devfiles to reuse global attributes in different sections of the same devfile.
+
+.Devfile variable
+
+The devfile 2.x specification includes global variables which define key value pairs.
+
+```yaml
+variables:
+  my-var: "some value"
+```
+
+Use `{{my-var}}` to reference the variables in non-string fields within the devfile. When you set only the variable field, you can make different template versions of your devfile.
+
+.Top-level attribute
+
+Use the top-level attribute to define attributes as implementation-dependent, free-form yaml attributes on the top level in a devfile. Use `{{attribute-name}}` to refer to the attribute with the key `attribute-name`. You can only resolve attributes at the top-level of the devfile. If you do not resolve variables in your devfile, then you receive a non-blocking warning.
```

Comment:
1. Use `{{attribute-name}}` to refer to the attribute with the key `attribute-name`.
2. Resolve the variables in your devfile to prevent a non-blocking warning. 
3. Add attributes to the `top-level` and `variables` field to support merging and overwriting.
4. See the following example of top-level attributes:
```yaml
schemaVersion: 2.0.0
metadata:
  name: java-maven
  version: 1.1.1
attributes:
  javaVersion: 11
components:
  - name: tools
    container:
      image: quay.io/eclipse/che-java{{javaVersion}}-maven:nightly
```